httpcore/backends/mock.py

```diff
@@ -68,8 +72,11 @@ class AsyncMockStream(AsyncNetworkStream):
     def __init__(self, buffer: typing.List[bytes], http2: bool = False) -> None:
         self._buffer = buffer
         self._http2 = http2
+        self._closed = False
 
     async def read(self, max_bytes: int, timeout: Optional[float] = None) -> bytes:
+        if self._closed:
+            raise ReadError("Connection closed")
```

Comment:
Since `read` and `close` are async and could run concurrency do you need to use a lock here and in `aclose`?

Reply:
There's no `await` operations here, so no context switching can happen while `read()` is running.

Reply:
Interesting, I hadn't thought about that — makes sense though!